### PR TITLE
User can buy a subscription

### PR DESCRIPTION
**Pull request by clalil**
# User can buy a subscription
[PT-link](https://www.pivotaltracker.com/story/show/169253777)

## This PR contains
- Stripe implemented on the client side (matching errors and routes on the server side)
- Configure navbar with payment form
- Add feature test with both happy and two different sad paths
- Basic styling

## What I have learned
- How to extract steps in cypress into a separate command module using JS objects and functions
- How to implement Stripe on the client side
- How to add authentication for headers using localStorage in the browser
- Further understanding of headers
- How to find an iframe when using Cypress by disabling chrome websecurity

## Image of Payment Form
Screenshot 2019-11-05 at 18 21 18

### CODE REVIEW

File: src/Components/SignUp.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ class SignUp extends Component {
         console.log('You have successfully been signed up')
       )
       .catch(error => {
+        debugger;
```
Comment (clalil) [inline src/Components/SignUp.js]: Thank you! Removed!

**Review by shsteindorsson — APPROVED** [review]

File: cypress/fixtures/successful_payment.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "message": "Transaction successful"
```
Comment (oliverochman) [inline cypress/fixtures/successful_payment.json]: indentation :)

File: cypress/integration/LoggedInUserCanBuyASubscription.spec.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  })
+
+  it('successfully signs up and enters the valid credentials', () => {
+    cy.route({
+      method: 'POST',
+      url: 'http://localhost:3000/v1/payments',
+      response: 'fixture:successful_payment.json',
+      status: 200
+    })
+    cy.signup_user(
```
Comment (oliverochman) [inline cypress/integration/LoggedInUserCanBuyASubscription.spec.js]: please use `camelCase` when writing methods in js.

File: src/App.js [hunk 4]
```diff
@@ -24,7 +25,9 @@ class App extends Component {
         <Route exact path='/signup' component={SignUp}>
           {this.props.currentUser.isSignedIn ? <Redirect to="/" /> : <SignUp />}
         </Route>
-
+        <Route exact path='/payment' component={PaymentForm}>
+          {this.props.currentUser.isSignedIn ? <PaymentForm /> : <Redirect to="/" />}
```
Comment (oliverochman) [inline src/App.js:29]: research `import { generateRequireSignInWrapper } from 'redux-token-auth';`

tex
https://github.com/CraftAcademy/Glocal_news-client/blob/development/src/App.js

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/fixtures/successful_payment.json (hunk 2 above)
Comment (clalil) [reply cypress/fixtures/successful_payment.json]: Solved!

File: cypress/integration/LoggedInUserCanBuyASubscription.spec.js (hunk 3 above)
Comment (clalil) [reply cypress/integration/LoggedInUserCanBuyASubscription.spec.js]: Of course!

File: src/App.js (hunk 4 above)
Comment (clalil) [reply src/App.js:29]: Of course!

Comment (clalil) [reply src/App.js:29]: Update: we found this was out of scope since the PR is about payment and we would have to change the entire navbar.

**Review by oliverochman — APPROVED** [review]

**Pull request merged**